### PR TITLE
cast to (int) for Curl adapter timeout config

### PR DESCRIPTION
**Pull request by samsonasik**
same as what applied into Socket adapter

### CODE REVIEW

**Comment by Xerkus** [comment]
What is the reason for accepting non-integer timeout values?
I would say exception should be thrown instead.

**Comment by samsonasik** [comment]
just for consistent with socket adapter. if it should be get exception, the in socket adapter should get exception as well?

**Comment by Xerkus** [comment]
That would be my preference, yes. I will need to look into it sometime later, may be i am not seeing something.

File: src/Client/Adapter/Curl.php [hunk 1]
```diff
@@ -203,9 +203,9 @@ public function connect($host, $port = 80, $secure = false)
         }
 
         if (isset($this->config['connecttimeout'])) {
-            $connectTimeout = $this->config['connecttimeout'];
+            $connectTimeout = (int) $this->config['connecttimeout'];
```
Comment (weierophinney) [inline src/Client/Adapter/Curl.php]: I would prefer the following approach:

- Grab the config value.
- If it is not an integer or not a numeric string, throw an `InvalidArgumentException`.
- Cast to integer.

That approach ensures strings like `timeout` do not get cast to `0`, which is obviously incorrect.

Comment (samsonasik) [reply src/Client/Adapter/Curl.php]: do we need to check `$this->config['timeout']` and $this->config['connecttimeout'] in `setOptions()` function ? and change it at Socket adapter as well?

Comment (samsonasik) [reply src/Client/Adapter/Curl.php]: I've updated with integer and numeric check, applied to socket adapter as well.

**Comment by weierophinney** [comment]
Thanks, @samsonasik!

**Pull request merged**